hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -382,10 +382,36 @@ public static GenericRecord rewriteRecord(GenericRecord oldRecord, Schema newSch
     return newRecord;
   }
 
+  public static GenericRecord rewriteRecord(GenericRecord genericRecord, Schema newSchema, boolean copyOverMetaFields, GenericRecord fallbackRecord) {
+    GenericRecord newRecord = new GenericData.Record(newSchema);
+    boolean isSpecificRecord = genericRecord instanceof SpecificRecordBase;
+    for (Schema.Field f : newSchema.getFields()) {
+      if (!isSpecificRecord) {
+        copyOldValueOrSetDefault(genericRecord, newRecord, f);
+      } else if (!isMetadataField(f.name())) {
+        copyOldValueOrSetDefault(genericRecord, newRecord, f);
+      }
```

Comment:
I think there was some bug with this logic which was fixed in #4789 . Can we keep this block of if-else consistent with what we have in `rewriteRecord(GenericRecord oldRecord, Schema newSchema)`?

Reply:
from what I see, it was some minor optimization. two if blocks were combined to one in 4789. not sure if there any bug in rewriteRecord method as such. Or are you talking about some other fix in the caller?